test/graph_rewrites/test_ng_var_update_ng_tensor.cc

```diff
@@ -40,85 +41,125 @@ namespace tensorflow {
-  // which is the desired configuration for the test
+  auto accum = ops::Variable(root.WithOpName("accum"), varShape, DT_FLOAT);
+  auto init_value2 = ops::Const(root, {{3.f, 3.f}, {3.f, 3.f}});
+  auto accum_assign =
+      ops::Assign(root.WithOpName("Assign2"), accum, init_value2);
 
-  // Create NGraphVariableUpdateNGTensor node
-  ASSERT_OK(NodeDefBuilder("sync_node", "NGraphVariableUpdateNGTensor")
-                .Input(FakeInput(DT_FLOAT_REF))
-                .Attr("T", DT_FLOAT)
-                .Attr("ngraph_variable_shared_name", "var1")
-                .Attr("ngraph_graph_id", 1)
-                .Finalize(node_def()));
-  ASSERT_OK(InitOp());
+  auto grad = ops::Const(root, {{2.f, 2.f}, {2.f, 2.f}});
 
-  // Add NGraph resource to the same container as the test op
-  ContainerInfo cinfo_;
-  NodeDef ndef;
-  ndef.set_name("node1");
-  AddNodeAttr("container", "", &ndef);
-  AddNodeAttr("shared_name", "var1", &ndef);
-  ASSERT_OK(cinfo_.Init(device_->resource_manager(), ndef, true));
+  auto lr = ops::Const(root, 1.f);
 
-  // The Create function used here does not use Ref, thus does not
-  // increase the reference count for the resource. Therefore, we do not
-  // need to Unref before exiting.
-  ASSERT_OK(device_->resource_manager()->Create<NGraphVar>(cinfo_.container(),
-                                                           cinfo_.name(), var));
+  ops::ApplyAdagrad::Attrs use_locking;
+  use_locking = use_locking.UseLocking(true);
+  auto applyadagrad_t = ops::ApplyAdagrad(root.WithOpName("Adagrad"), var,
+                                          accum, lr, grad, use_locking);
 
-  // lock_for_refs_ : Used as the Mutex for inputs added as refs
-  inputs_.push_back({&lock_for_refs_, var->tensor()});
+  // Turn off optimizations so that all the nodes are processed
+  tensorflow::SessionOptions options;
+  options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(tensorflow::OptimizerOptions_Level_L0);
+  options.config.mutable_graph_options()
+      ->mutable_rewrite_options()
+      ->set_constant_folding(tensorflow::RewriterConfig::OFF);
 
-  ASSERT_OK(RunOpKernel());
+  // Run on nGraph
+  ActivateNGraph();
+  ClientSession ng_session(root, options);
+  std::vector<tensorflow::Tensor> ng_outputs1;
+  std::vector<tensorflow::Tensor> ng_outputs2;
+  ASSERT_OK(ng_session.Run({{var_assign, accum_assign}}, &ng_outputs1));
 
-  shared_ptr<ngraph::runtime::Tensor> ng_t = var->ng_tensor();
-  Tensor output_tensor(DT_FLOAT, TensorShape({2}));
-  void* dst_ptr = DMAHelper::base(&output_tensor);
-  ng_t->read(dst_ptr, 0, output_tensor.TotalBytes());
+  ASSERT_OK(ng_session.Run({applyadagrad_t}, &ng_outputs2));
 
-  Compare(output_tensor, input_tf_tensor, 0);
+  DeactivateNGraph();
 
+  // Run on TF
+  ClientSession tf_session(root, options);
+  std::vector<tensorflow::Tensor> tf_outputs1;
+  std::vector<tensorflow::Tensor> tf_outputs2;
```

Comment:
The end to end test can be added to test_variables.cc or a new file outside this folder. We can use this for only testing the graph rewrites.